scripts/asm-installer/install_asm

```diff
@@ -661,10 +668,9 @@ EOF
 validate_expected_control_plane(){
   info "Checking Istio installations..."
   check_no_istiod_outside_of_istio_system_namespace
-  if [[ "${MODE}" = "migrate" ]]; then
+  if [[ "${MODE}" = "migrate" || "${MODE}" = "upgrade" ]]; then
```

Comment:
This can be a bit more succinct.

`if [[ "${MODE}" = @(migrate|upgrade) ]]; then` should work w/pattern matching.

Reply:
SG

Reply:
posh compliance fails with:

`if [[ "${MODE}" = @(migrate|upgrade) ]]; then` any thoughts?

Reply:
Nope, posh is right, that's a bashism. Your original way is better.